LPTMR counter prescaler fix

FelixWang47831 opened the pull request:
For previous code, if timer_mode_sel is set to 0(Time Counter Mode), bypass_prescaler_glitch will be 1-0 = 1, which means the prescaler does not work on time counter mode. But prescaler is very useful, especially for long timing count.

In this PR:
1. Update  prescale_glitch_filter in dts bindings,  add 0 in enum and the default value for this property has been changed to 0, to indicate  prescaler and glitch filter are disabled. 
  Have checked all of device tree settings, the prescaler property are all 1, which means the clock are not prescaled. And in previous driver code,  bypass_prescaler_glitch are always 1, the prescaler never works. Therefore this change will not bring running issue for existing boards.

2. If user sets prescale-glitch-filter properity other than 0, set bypass_prescaler_glitch to false to make prescaler work.

3. Calculate clock frequency with prescaler setting

FelixWang47831 committed:
tests: drivers: counter_basic_api: Enable lptmr test for mimxrt1180_evk

The lptmr0 has been enabled with default settings defined in
dts/arm/nxp/nxp_rt118x.dtsi. Enable lptmr3 to test with
designated prescaler.

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

FelixWang47831 committed:
drivers: Counter: Fix LPTMR prescaler setting not work issue

In current code, if timer-mode-sel is 0(Timer Counter Mode),
bypass_prescaler_glitch will be 1, that makes prescaler setting
be bypassed. But this setting is very useful, especially for
long timing count.

In this patch, we update prescale-glitch-filter default value to 0,
to indicate prescaler and glitch filter are disabled, which comply
the existing devices DTS setting (prescaler = 1).
And if user sets prescale-glitch-filter properity other than 0,
we should set bypass_prescaler_glitch to false to make prescaler work,
and the clock frequence should be calculated with prescaler setting.

Update prescaler field in dt-bindings to tell developer should use
prescale-glitch-filter instead.

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

Holt-Sun commented on drivers/counter/counter_mcux_lptmr.c:
```diff
@@ -178,20 +178,22 @@ static DEVICE_API(counter, mcux_lptmr_driver_api) = {
 				GENMASK(DT_INST_PROP(n, resolution) - 1, 0),	\
-			.freq = DT_INST_PROP(n, clock_frequency) /		\
-				DT_INST_PROP(n, prescaler),			\
+			.freq = DT_INST_PROP(n, clock_frequency) / (1 <<	\
```
Use BIT()
```suggestion
            .freq = DT_INST_PROP(n, clock_frequency) /		\
                BIT(DT_INST_PROP(n, prescale_glitch_filter)),	\
```

FelixWang47831 replied:
Fixed

Holt-Sun commented on dts/bindings/counter/nxp,lptmr.yaml:
```diff
@@ -50,15 +50,19 @@ properties:
+        2 ^ [prescaler-glitch-filter] rising edges detected by the pin configured
+        from the input-pin value.
+      Note, that the pulse mode cannot be 2 ^ 16.
 
```
Should we deprecate the prescaler here?
```suggestion
  # Keep for DT ABI, but no longer used by the driver
  prescaler:
    type: int
    deprecated: true
    description: Deprecated; use prescale-glitch-filter instead.
```

FelixWang47831 committed:
dts: arm: nxp: Update lptmr prescaler setting

For lptmr counter driver, since "prescaler" has been
deprecated and replaced by prescale-glitch-filter.
Update related setting in device tree files.

Signed-off-by: Felix Wang <fei.wang_3@nxp.com>

dleach02 commented:
@FelixWang47831 need to address merge conflicts
